api/envoy/api/v2/route/route.proto

```diff
@@ -903,6 +903,26 @@ message RetryPolicy {
     google.protobuf.Duration max_interval = 2 [(validate.rules).duration = {gt {}}];
   }
 
+  message RetryBudget {
```

Comment:
Note, this feature needs both release notes and some discussion in the arch overview somewhere.